### ISSUE TITLE
Evaluate bounding boxes

### ISSUE THREAD

**Issue by bw4sz** [weecology/DeepForest-pytorch#16]
main.py deepforest class needs an evaluate method that returns

* mean-average-precision for a given intersection over union threshold between ground truth and predicted bounding boxes.
* precision at a fixed probability cutoff
* recall at a fixed probability cutoff

```
mAP, precision, recall = m.evaluate("/orange/b.weinstein/NeonTreeEvaluation/benchmark_annotations.csv", metrics=["mAP","precision","recall"], iou_threshold=0.4, probability_threshold=0.2)
```

Much of this code can be take and generalized from 

https://github.com/weecology/NeonTreeEvaluation_python/blob/eaa5dfb3d2a17daf354c146ab158f69b94755535/src/eval.py#L9

**Comment by bw4sz** [comment]
Need a sanitize file checker for format of input files

```
from deepforest.utilities import check_file
def check_file(df):
    """Check a file format for correct column names and structure"""
...
```

**Comment by bw4sz** [comment]
closed commit 0fd96831e4a650c164acf60befcc5010bc6538bd
